src/tools/processData.js

```diff
@@ -105,12 +105,41 @@ const convertEmails = (emails) => {
       const foundTransformedUser = transformedUsers.find((user) => user.id === userId);
       return foundTransformedUser ? foundTransformedUser.email : '';
     }
-    return '';
+    return emails.includes('@') ? faker.internet.email() : '';
   });
 
   return convertedEmails.join(', ');
 };
 
+const convertName = (name, email) => {
+  if (!name) {
+    return { name, email };
+  }
+  const additionalId = 99999;
+  // const emailsArray = emails.split(', ');
+  // const convertedEmails = emailsArray.map((email) => {
+  let foundUser = realUsers.find((user) => user.email === email);
+
+  // Not all program specialists or grant specialist are in the Hub yet
+  // Add it to the realUsers
+  if (!foundUser && email.includes('@')) {
+  // if (!foundUser) {
+    foundUser = { id: additionalId + 1, name, email };
```

Comment:
We save the specialists that don't exist in the memory with that additionalId. That way any new records that have the same specialist would reference the same data. We want to increment it for any next specialist that doesn't exist in the Hub and doesn't exist in memory yet.

Reply:
So I'm probably just not getting this - wouldn't this functionally always be
```js 
foundUser = { id: 10000, name, email };
```
? 

That is to say, wouldn't additionalId just always be created as 9999 each time the convertName function is called, and if ```!foundUser && email.includes('@')```, it would always be incremented and passed in as 10,000? Maybe thats what we want?

Reply:
> That is to say, wouldn't additionalId just always be created as 9999 each time

I would imagine that is what happens?